### ISSUE TITLE
VS complains about missing `android` workload even after resolving it at the command line

### ISSUE THREAD

**Issue by AArnott** [dotnet/maui#3923]
### Description

After installing the Maui optional component as part of the "Mobile development with .NET" workload in the VS installer, "dotnet workloads" were still missing (#3859). But this bug is about how VS gets stuck on the `android` dotnet workload being missing even after resolving it. 

Afte

### Steps to Reproduce

A clean install of VS with the Mobile development workload and Maui selected.
Create a new Maui app project.
Try to build. Fails due to #3859. In particular, VS produces this output:

```
Build started...
1>------ Build started: Project: MauiApp1, Configuration: Debug Any CPU ------
1>C:\Program Files\dotnet\sdk\6.0.200-preview.21573.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: android
1>C:\Program Files\dotnet\sdk\6.0.200-preview.21573.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload install android
1>Done building project "MauiApp1.csproj" -- FAILED.
1>You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
1>MauiApp1 -> C:\temp\MauiApp1\MauiApp1\bin\Debug\net6.0-windows10.0.19041\win10-x64\MauiApp1.dll
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

Note the instruction to run `dotnet workload install android`. 

After running `dotnet workload install android` at the CLI, the build in VS fails **with the same error and prescribed fix**. **This is the bug that *this* issue tracks.**.

In fact the CLI build also fails, until I run these several commands:

```
dotnet workload install android
dotnet workload install maui-android
dotnet workload install maccatalyst
```

Then finally the CLI build works. But _still_ the build in VS fails with the same output as above, telling me that the `android` workload is missing and to run the first command above, which has already run successfully.

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

VS 17.1.0 Preview 3 (32102.18.main)

### Did you find any workaround?

Building at the command line with `msbuild.exe` works. I can't find a way to get the build to succeed in VS.

### Relevant log output

```shell
Build started...
1>------ Build started: Project: MauiApp1, Configuration: Debug Any CPU ------
1>C:\Program Files\dotnet\sdk\6.0.200-preview.21573.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: android
1>C:\Program Files\dotnet\sdk\6.0.200-preview.21573.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload install android
1>Done building project "MauiApp1.csproj" -- FAILED.
1>You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
1>MauiApp1 -> C:\temp\MauiApp1\MauiApp1\bin\Debug\net6.0-windows10.0.19041\win10-x64\MauiApp1.dll
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

**Comment by marcpopMSFT** [comment]
@AArnott , after you installed android on the CLI the first time, were you able to build from the CLI? We think we identified a bug related to us not listing all workloads you need to install but once you installed android, you should have gotten a different error. We're wondering if somehow VS and the CLI are using different SDKs. If you hit this again, can you run dotnet --info and try to build on the CLI?

BTW, we're also going to change that install message to be a dotnet workload restore (and to list all versions of your missing workloads)

**Comment by AArnott** [comment]
Yes, ultimately I got the CLI build to work because the error message changed in the CLI build each time I resolved something. All the while VS still complained with the original error.

**Comment by pierre01** [comment]
I installed and uninstalled VS 2022 preview 4 multiple times but the compiler keeps on  telling :

Build started...
1>------ Build started: Project: MauiApp2, Configuration: Debug Any CPU ------
1>C:\Program Files\dotnet\sdk\6.0.200-preview.22055.15\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: maui-android
1>C:\Program Files\dotnet\sdk\6.0.200-preview.22055.15\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload install maui-android
1>Done building project "MauiApp2.csproj" -- FAILED.
1>C:\Program Files\dotnet\sdk\6.0.200-preview.22055.15\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: maui-ios
1>C:\Program Files\dotnet\sdk\6.0.200-preview.22055.15\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload install maui-ios
1>Done building project "MauiApp2.csproj" -- FAILED.
1>C:\Program Files\dotnet\sdk\6.0.200-preview.22055.15\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: maui-maccatalyst
1>C:\Program Files\dotnet\sdk\6.0.200-preview.22055.15\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload install maui-maccatalyst
1>Done building project "MauiApp2.csproj" -- FAILED.
1>C:\Program Files\dotnet\sdk\6.0.200-preview.22055.15\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: maui-windows
1>C:\Program Files\dotnet\sdk\6.0.200-preview.22055.15\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload install maui-windows
1>Done building project "MauiApp2.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

I can't compile any  maui programs and I tried to reinstall the workloads using tthe CLI and MAUI-check to no avail
I am running on windows 11.

I did a full VS Install cleanup, it did not work either..
